inventory/main/src/main/java/org/opennms/horizon/inventory/monitoring/MonitoredEntityService.java

```diff
@@ -35,15 +36,22 @@
 @Service
 @RequiredArgsConstructor
 public class MonitoredEntityService {
-    private final List<MonitoredEntityProvider> providers;
 
+    private final List<MonitoredEntityProvider> providers;
+    private final MonitoringLocationRepository monitoringLocationRepository;
     private final TaskSetPublisher taskSetPublisher;
 
     public Stream<MonitoredEntity> getAllMonitoredEntities(final String tenantId, final long locationId) {
         return this.providers.stream()
                 .flatMap(provider -> provider.getMonitoredEntities(tenantId, locationId).stream());
     }
 
+    public Stream<MonitoredEntity> getAllMonitoredEntities(final String tenantId) {
+        return monitoringLocationRepository.findByTenantId(tenantId).stream()
+                .flatMap(location -> this.providers.stream()
+                        .flatMap(provider -> provider.getMonitoredEntities(tenantId, location.getId()).stream()));
```

Comment:
You can optimize by swapping:

```suggestion
        return this.providers.stream()
            .flatMap(provider -> monitoringLocationRepository.findByTenantId(tenantId).stream()
                .map(location -> provider.getMonitoredEntities(tenantId, location.getId()));
```

Reply:
Done.